src/isomorphic/ReactLifeCycleTimers.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactLifeCycleTimers
+ */
+
+'use strict';
+
+var keyMirror = require('keyMirror');
+
+var ReactLifeCycleTimers = keyMirror({
+  COMPONENT_DID_MOUNT: null,
+  COMPONENT_DID_UPDATE: null,
+  COMPONENT_WILL_MOUNT: null,
+  COMPONENT_WILL_RECEIVE_PROPS: null,
+  COMPONENT_WILL_UNMOUNT: null,
+  COMPONENT_WILL_UPDATE: null,
+  CONSTRUCTOR: null,
+  UPDATE_REFS: null,
+  RENDER: null,
+  SHOULD_COMPONENT_UPDATE: null,
+});
```

Comment:
Personally, I'm not a fan of explicit runtime enums. I like the Flow version better where the type can be validated but you don't have import a third party module to do it.

It's worse for perf than a regular string. These short strings should be interned and inlined anyway. This just makes it harder for a VM or compiler to do that inlining since these objects are mutable. It's not safe to inline.

keyMirror is also tricky in regard to property name mangling. E.g. if we want to mangle property names inside a React compilation, rather than whole program. (Two separate compile steps.)

We should get rid of this pattern completely.

More importantly, this is the first time the value leaks into a public API. If we change convention or something internally, it would affect the public API.

Instead I'd rather just put a string inline at each callsite. Like an event.

Given that, what should the event convention be? Does a camel-case "componentDidMount" make more sense given that is what the component method is called?